lib/datadog/profiling/stack_recorder.rb

```diff
@@ -0,0 +1,31 @@
+# typed: false
+
+module Datadog
+  module Profiling
+    # Used to wrap a ddprof_ffi_Profile in a Ruby object and expose Ruby-level serialization APIs
+    # Methods prefixed with _native_ are implemented in `stack_recorder.c`
+    class StackRecorder
+      def serialize
+        status, result = self.class._native_serialize(self)
+
+        if status == :ok
+          start, finish, encoded_pprof = result
+
+          [start, finish, encoded_pprof]
```

Comment:
This is a bit redundant, but on purpose -- I wanted to have a Ruby-level statement of what comes out of the native code.

Reply:
There's performance impact from doing this (the array creation), would it make sense to document this instead?

Reply:
I don't think it makes a measurable difference in this case:

* This code is expected to run once every 60 seconds
* Since the new array is 3 items long, it doesn't even need a malloc -- [it just fits fully within a struct-RVALUE sized slot](https://ivoanjo.me/blog/2021/02/11/looking-into-array-memory-usage/#whats-in-a-ruby-object)